### ISSUE TITLE
Command Interpolation Enhancements

### ISSUE THREAD

**Issue by stoutput** [bibendi/dip#118]
2 Enhancement Requests:

1. **Do not replace non-environment variables in provisioning commands.**
    For example, in this `dip.yml`: ![](https://i.imgur.com/NmYXRws.png) The first provision command outputs: 
    ```
    git@github.com:organization/repo.git git@github.com:organization/repo2.git
    repo name:
    repo name:
    ```
    I've debugged it quite thoroughly and it seems that the interpolation step [here]( https://github.com/bibendi/dip/blob/1a06aa9cf448854a77e0547cd71814e54f6f392b/lib/dip/command.rb#L26) and subsequently [here](https://github.com/bibendi/dip/blob/1a06aa9cf448854a77e0547cd71814e54f6f392b/lib/dip/environment.rb#L37) replaces all variables with their corresponding environment variable first from the yml, then the environment if it exists, else it gets replaced with a blank string (hence why `$REPOS` is correctly replaced, but `$repo` is blank). This should instead _**only**_ replace the variable if an environment variable has been **explicitly declared**.

    This would allow one to offload a good bit of setup logic into dip's provisioning functionality. I have this dip.yml at the root of my projects, and a simple `dip provision` could bring everything up for me without having to create a bunch of shell scripts, if this were to work.

2. **Allow generalized, folder-specific command aliases**
    One of my favorite features of dip is its ability to dynamically alias commands when a dip.yml file is found. I'd like to be able to add additional aliases that do not reference a specific docker-compose service. In the example above, you can see I've stubbed out what I'd imagine it to look something like under "interaction". Currently running this command results in a `lib/dip/interaction_tree.rb:61:in 'fetch': key not found: :service` error since the service key is not found.
    
    I would love to be able to use dip for all my project-specific aliases, and not just docker-compose ones (the goal is for it to feel native, right? 💯 ). If this were to work, I could run a simple `dip start` to bring up both nginx and dns, as well as anything else necessary for the environment to function!

P.S. While I'm not a Ruby dev, I'm loving dip so far. It's encapsulated a lot of my ideal workflow and tooling when working in a multi-project Docker environment. Great work on this! I'm hoping to be able to adopt this or a flavor of it into our company's workflow. I'd love to see some more documentation around the capabilities of it if you ever find the time!

**Comment by stoutput** [comment]
I'm no Ruby dev, but I believe the `else` [here](https://github.com/bibendi/dip/blob/1a06aa9cf448854a77e0547cd71814e54f6f392b/lib/dip/environment.rb#L43) should be replaced with something like
```
elsif defined? self[var_name]
    self[var_name]
else
    #return original value ($var_name)
end
```
To fix enhancement # 1.

**Comment by bibendi** [comment]
Hi!
Thank you for your suggestions! 
Enhancement # 1 looks fixable very easily.
Number 2 may be implemented after I will complete the #113 feature. I'm a little bit swamped for now, so progress on it completed just only by 70%. I will try to go faster :)

**Comment by stoutput** [comment]
Thanks for your work on this!
